environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/event/options/mode/NotificationModeTest.java

```diff
@@ -93,15 +95,15 @@ public void testExceptionPropagated() throws InterruptedException {
 
     @Test
     public void testInvalidMode() throws InterruptedException {
-        Set<String> foos = new CopyOnWriteArraySet<>();
+        CountDownLatch latch = new CountDownLatch(1);
         try (WeldContainer container = createWeld()) {
             try {
-                container.event().select(Message.class).fireAsync(() -> foos.add("bar"), NotificationOptions.of(WeldNotificationOptions.MODE, "unsupported"));
+                container.event().select(Message.class).fireAsync(() -> latch.countDown(), NotificationOptions.of(WeldNotificationOptions.MODE, "unsupported"));
                 fail("Notification should have failed ");
             } catch (IllegalArgumentException expected) {
             }
             // Assert that observers were not notified
-            assertTrue(foos.isEmpty());
+            assertFalse(latch.await(500, TimeUnit.MILLISECONDS));
```

Comment:
This looks good. Or at least I am sure I can't think of anything faster right now :-)